compiler/typecheck/TcErrors.hs

```diff
@@ -2404,6 +2405,28 @@ mkDictErr ctxt cts
     -- but we really only want to report the latter
     elim_superclasses cts = mkMinimalBySCs ctPred cts
 
+customDamlErrors :: Ct -> [ClsInst] -> SDoc -> Maybe SDoc
+customDamlErrors ct candidate_insts binds_msg
+  | TyConApp con [TyConApp target [], viewType] <- ctev_pred (ctEvidence ct)
+  , "HasInterfaceView" <- occNameString $ occName $ tyConName con
```

Comment:
I think we've discussed this in the past, it's very unlikely that a user will write their own class `HasInterfaceView` with two arguments and because of this code get a confusing error message, but out of principle I think we should follow GHC's approach of assigning keys to "known" names, see [`compiler/prelude/PrelNames.hs`](https://github.com/digital-asset/ghc/blob/da-master-8.8.1/compiler/prelude/PrelNames.hs). This is not a blocker, though, we could just open an issue and leave some `TODO`s around.

Reply:
fixed